pkg/runtime/internal/envdef/environment.go

```diff
@@ -82,6 +82,19 @@ func (j *VariableJoin) MarshalText() ([]byte, error) {
 	return []byte(res), nil
 }
 
+func (j VariableJoin) MarshalJSON() ([]byte, error) {
```

Comment:
I need this because `MarshalText()` was not being called by the JSON marshaler, resulting in integers (not strings) being written to our injected runtime.json files.